### PR TITLE
Add 'Create IoTjs Tizen Project' option to extension

### PR DESCRIPTION
**Pull request by tdusnoki**
IoT.js-VSCode-DCO-1.0-Signed-off-by: Tibor Dusnoki tdusnoki@inf.u-szeged.hu

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -205,7 +205,10 @@ After installing Tizen Studio you can add the following lines of information to
     "IoTjsPath": "/absolute/path/to/iotjs"
 }
 ```
-This enables the extension to install required packages for Tizen Studio to be able to create IoTjsApp native project. The installation may take several minutes. 
+This enables the extension to install required packages for Tizen Studio to be able to create[IoTjsApp native project](#create-tizen-native-project). The installation may take several minutes.
```
Comment (robertsipka) [inline README.md]: A space is missing before the name of the relative link.

File: tools/CreateTizenProject.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# usage: CreateTizenProject.sh [projectName] [/path/to/destination/directory] [/path/to/tizen/studio]
+
+[ "$#" -ne 3 ] && { echo "Usage: $0 <project-name> <destination-directory> <tizen-studio>"; exit 1; }
+
+TIZEN_PROJECT=$1
+PROJECT_DIRECTORY=$2
+TIZEN_STUDIO=$3
+export PATH=$PATH:$TIZEN_STUDIO/tools/ide/bin
+tizen create native-project -p iot-headless-4.0 -t IoTjsApp -n $TIZEN_PROJECT -- $PROJECT_DIRECTORY
+tizen build-native -r iot-headless-4.0-device.core -- $PROJECT_DIRECTORY/$TIZEN_PROJECT
```
Comment (robertsipka) [inline tools/CreateTizenProject.sh]: A newline is missing at the end of file.

File: src/extension.ts [hunk 3]
```diff
@@ -62,6 +68,51 @@ const provideInitialConfigurations = (): string => {
...
+    ignoreFocusOut: true
+  });
+  if (projectName && tizenStudioPath) {
+    const projectPath = await vscode.window.showOpenDialog({
+      openLabel: 'Set as destination directory',
+      canSelectFiles: false,
+      canSelectFolders: true,
+      canSelectMany: false
+    });
+    if (projectPath) {
```
Comment (robertsipka) [inline src/extension.ts:98]: Is it possible to not select a folder with showOpenDialog? If not, the projectPath always has a value and this check is unnecessary.

**Review by robertsipka — CHANGES_REQUESTED** [review]

File: src/extension.ts (hunk 3 above)
Comment (tdusnoki) [reply src/extension.ts:98]: You can cancel the dialog window and in that case projectPath is undefined and you can only create the project if you have a valid projectPath.

Comment (rtakacs) [reply src/extension.ts:98]: Could you check the `tizenStudioPath` at the begin of the `createTizenProject` function? You could show a pop-up window (with some information) if the `tizenStudioPath` is not specified.

File: src/extension.ts [hunk 4]
```diff
@@ -184,31 +237,42 @@ const createHover = (document: vscode.TextDocument, position: vscode.Position):
 };
 
 const setup = () => {
+  const setupPath = path.join(__dirname, 'setup.sh');
+  if (tizenStudioPath && IoTjsPath) {
```
Comment (rtakacs) [inline src/extension.ts]: Please check the `tizenStudioPath` and `IoTjsPath` variables at the beginning of the `setup` function.

File: package.json [hunk 5]
```diff
@@ -77,15 +77,30 @@
...
   "main": "./out/extension.js",
   "files": [
     "out/**"
   ],
   "contributes": {
+    "commands":[ {
+      "command": "iotjs-debug.createTizenProject",
+      "title": "Create IoTjs Tizen Project"
+    }
+
```
Comment (rtakacs) [inline package.json]: Unnecessary empty line.

File: src/extension.ts [hunk 6]
```diff
@@ -62,6 +68,53 @@ const provideInitialConfigurations = (): string => {
   ].join('\n');
 };
 
+const createTizenProject = async(): Promise<any> => {
+  fs.stat(lastModified.filePath, (err, stat) => {
+    if (lastModified.mtime < stat.mtime) {
+      getPath();
+      setup();
```
Comment (rtakacs) [inline src/extension.ts]: Please remove this line since the `getPath` function executes the `setup` function.

File: src/extension.ts [hunk 7]
```diff
@@ -62,6 +68,60 @@ const provideInitialConfigurations = (): string => {
...
 };
 
+const checkPath = () => {
+  fs.stat(lastModified.filePath, async (err, stat) => {
+    if (lastModified.mtime < stat.mtime) {
+      await setup();
+      createTizenProject();
+    } else {
+      createTizenProject();
+    }
```
Comment (robertsipka) [inline src/extension.ts]: The else case is unnecessary. You can call the createTizenProject after the if statement.

**Review by robertsipka — APPROVED** [review]
LGTM

**Review by rtakacs — APPROVED** [review]
LGTM

**Pull request merged**